### PR TITLE
Stim Medipen for mining

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

In this PR I introduce a new medipen, containing stim. mix!

## Why It's Good For The Game

Miners can't live with only legion cores, so I added this medipen for 750 credits

## Changelog
:cl:
add: Added new medipen containing stimulants
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Do miners need to be more combat capable than they already are?

**Comment by SmArtKar** [comment]
> Do miners need to be more combat capable than they already are?

Yeah, now they have no speedboosts, only legion cores that are meh.

**Comment by bobbahbrown** [comment]
consider making this not work on station

**Comment by boodaliboo** [comment]
> consider making this not work on station

are you asking for the chemicals to be changed or just the pen

**Comment by ExcessiveUseOfCobblestone** [comment]
yes they can. if you are that hard pressed bother chemistry.

**Comment by ATH1909** [comment]
decent idea, but WOW, this is not the way to do this.

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 1]
```diff
@@ -243,6 +243,15 @@
...
 	list_reagents = list(/datum/reagent/medicine/salbutamol = 10, /datum/reagent/medicine/c2/penthrite = 10, /datum/reagent/medicine/oxandrolone = 10, /datum/reagent/medicine/sal_acid = 10 ,/datum/reagent/medicine/omnizine = 10 ,/datum/reagent/medicine/leporazine = 10)
 
+/obj/item/reagent_containers/hypospray/medipen/stimpack/mining //REAL goliath kiting
+	name = "emergency stimpack medipen"
+	desc = "A rapid way to inject some stimulants to move faster while killing local fauna."
+	icon_state = "stimpen"
+	inhand_icon_state = "stimpen"
+	volume = 30
+	amount_per_transfer_from_this = 30
+	list_reagents = list(/datum/reagent/medicine/ephedrine = 8, /datum/reagent/consumable/coffee = 8, /datum/reagent/medicine/epinephrine = 8, /datum/reagent/medicine/morphine = 6)
```
Comment (ATH1909) [inline code/modules/reagents/reagent_containers/hypospray.dm:253]: this pen is awful awful awful awful for actual fighting on lavaland, and even in PvP, it stinks

Ephedrine can and will make you drop your PKA several times, possibly mid-combat. Coffee and epinephrine will react with each other to form pump-up, a chem with the sole upside of giving you the TRAIT_STUNRESISTANCE trait... which ONLY affects the lengths of knockdowns from stun batons (not any other source of stuns), and is _already provided by the ephedrine anyway_. The epinephrine present in this pen will also react with the penthrite from a luxury survival medipen, causing you to explode if you try to use both of them together (even though this medipen's intended purpose seems to be to be used alongside luxury survival medipens while fighting megafauna). Finally, the morphine in this medipen is present in a large enough dose to guarantee that you'll be knocked out by it (the ephedrine won't even help you with waking up from the chain knockouts after tick 24-ish, because from that point on, morphine will increase your sleeping duration to 4 seconds every 2 seconds, while ephedrine will merely reduce it by 2 seconds every 2 seconds, and the coffee will be long gone by then), which is, of course, a death sentence during a megafauna fight.

In short, literally every chem in this pen actively harms your ability to fight (mega)fauna on lavaland.

**Comment by ATH1909** [comment]
image

oh no

**Comment by LemonInTheDark** [comment]
To elaborate on ath's picture, the master branch is supposed to mirror the thing you're pushing to, before your push. That's a simplified description of things, ask about it in coding general for more details.

Oh and give this man a good first pr for the love of god

**Comment by challade** [comment]
> 
> 
> Do miners need to be more combat capable than they already are?

I haven't seen anyone been able to combat bubblegum for a while now without being borged (heat resistant, self heal, no slow). Maybe there's some guy out there that's been able to fight bubblegum, but it literally requires not getting hit at any point whatsoever. This is all in regards of the PkA. I can't even imagine anyone even considering using the crusher. So yes, miners need to be combat ABLE. Stop with this new policy of actively making things worse, **if there's a problem on lavaland consider ADDING a way to fix the problem rather than removing the problem altogether.** 


This pull request is a step in the right direction. The item dropping is not the way to go about it though, either create a new chem that only works on lavaland/icemoon or change how these chems work there because there is no way anyone's going to buy that stem pack except newbies who are gonna be confused as to what's happening.
 Stop the newbie abuse. All these lavaland hating fucks already made it so that if a newbie rations his points by buying different types of med pens (survival+luxury) they explode and die. I learned this the hard way when fighting a megafauna and didn't know about the new "feature".

**Comment by Ratvar** [comment]
Instead of buffing miners, you can nerf bubblegum and rewards

**Comment by anconfuzedrock** [comment]
This is like the zip 22 of mining pens

**Comment by challade** [comment]
> Instead of buffing miners, you can nerf bubblegum and rewards, and go fix megafauna in general.

I see your point. The rewards for Bubblegum are really good all things considered, maybe the blood contract merits the risk of dying to BBG easily. But, I don't think removing it is the way to go. Lots of people enjoy playing lavaland for the risk and thrill of kicking ass, removing it serves no positive purpose. 
Honestly, it seems like bucket crab logic, "If some people can do it and not most, lets nerf it and lower the reward and risk so that more people are able to do it!" I disagree with this statement, it's equity at it's worst and it will only turn people AWAY from megafauna hunting and to some degree away from SS13 in general. Lavaland is it's own game within SS13 and it's a really good alternative to the boring grind that SS13 sometimes becomes.

I propose that rather than nerfing megafauna we instead buff some miner items (so long as it stays on lavaland). For example, maybe there could be some healing aspect with a crusher trophy that's somewhat rare but necessary to combat BBG.

Or perhaps a buff miners get only when they combat megafauna. Beast mode maybe? When you get into combat with megafauna you don't slow down and you have a higher crit threshold and you get the slightest of heal if you kill the fauna (that way you don't immediately fall over dead upon the buff running it's course after the battle).

Maybe even adding more variety to miners, we got PkA miners, Crusher hunters, what's left is a support role, someone who makes sure that if a miner is injured they get healing or they can fulton miners up (which is fine since most fauna gib miners). Comes with a very weak item that heals allies only on lavaland (requiring them to be on lavaland for the full heal)

With all this there might be more openings for megafauna variety, maybe some that FORCE miners to work together. Instead of turning lavaland into boring mining simulator, perhaps it can be turned into Monster Hunter: SS13 G.

These might sound too powerful, but all things considered fighting megafauna is no easy feat on its own. Plus, I'm more in favor of making things fun and awesome than making everything boring and simplified.

**Comment by Doosteroni** [comment]
> I propose that rather than nerfing megafauna we instead buff some miner items

Like say, reverting a crappy PR that nerfed  one of the only things making megafauna hunting possible, one that was born out of the issue of mining pens being used on station but for some god forsaken reason nerfed the medipen in the first place? Just a though

**Comment by ExcessiveUseOfCobblestone** [comment]
> 
> 
> yes they can. if you are that hard pressed bother chemistry.
